src/api/hotel/hotel.controller.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response } from "express";
+
+import { createHotel } from "./hotel.service";
+import { Hotel, CreateHotelData } from "./hotel.types";
+
+export async function createHotelController(req: Request, res: Response){
+    try{
+        const hotelData: CreateHotelData = req.body; 
+        const createdHotel: Hotel = await createHotel(hotelData);
+        res.status(201).json({ message: 'hotel has been created successfully',createdHotel});
+    } catch ({ message }: any) {
```

Comment:
I recommend create an importable resource for error management which contains a function like this:
```ts
function errorHandler(exception: unknown) {
  const message = (typeof exception === 'string') ? exception.toUpperCase()
    : (exception instanceof Error) ? exception.message : 'Something was wrong';

  return message;
}

export default errorHandler;
```

Then, in try catch implement something like this:
```ts
...
} catch(error: unknown) {
 const message = errorHandler(error);
 res.status(400).json({ message });
}
```

And do not forget semicolon once you use it in a project, thx.

Reply:
Apply this for a better error exception management instead of using any.